src/core/processing/qgsprocessingregistry.cpp

```diff
@@ -156,9 +156,17 @@ bool QgsProcessingRegistry::removeProvider( const QString &providerId )
   return removeProvider( p );
 }
 
-QgsProcessingProvider *QgsProcessingRegistry::providerById( const QString &id )
+QgsProcessingProvider *QgsProcessingRegistry::providerById( const QString &id ) const
 {
-  return mProviders.value( id, nullptr );
+  auto it = mProviders.constFind( id );
+  if ( it != mProviders.constEnd() )
+    return it.value();
+
+  // transparently map old references to "grass7" provider to "grass" provider
+  if ( id.compare( QLatin1String( "grass7" ), Qt::CaseInsensitive ) == 0 )
```

Comment:
This condition is required to run old scripts, right?

Reply:
Yep, and old models